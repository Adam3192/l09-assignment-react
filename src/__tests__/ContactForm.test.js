import { fireEvent, render, screen } from '@testing-library/react'
import ContactForm from '../ContactForm'

afterEach(() => {
  jest.restoreAllMocks()
})

it('renders a Add Contact button', () => {
  render(<ContactForm />)
  expect(screen.getByText('Add Contact')).toBeInTheDocument()
})

it('renders a Name form field', () => {
  render(<ContactForm />)
  expect(screen.getByRole('textbox', { name: 'Name' })).toBeInTheDocument()
})

it('renders a Email form field', () => {
  render(<ContactForm />)
  expect(screen.getByRole('textbox', { name: 'Email' })).toBeInTheDocument()
})

it('renders a Phone form field', () => {
  render(<ContactForm />)
  expect(screen.getByRole('textbox', { name: 'Phone' })).toBeInTheDocument()
})

it('verifies form value is updated when typing', () => {
  render(<ContactForm />)
  const input = screen.getByRole('textbox', { name: 'Name' })
  fireEvent.change(input, { target: { value: 'John' } })
  expect(input.value).toBe('John')
})

it('verifies callback function invoked when button clicked', () => {
  let onContactAdded = jest.fn()
  render(<ContactForm onContactAdded={onContactAdded} />)
  const button = screen.getByRole('button', { name: 'Add Contact' })
  fireEvent.click(button)
  expect(onContactAdded).toHaveBeenCalledTimes(1)
})

it('renders without error', () => {
  render(<ContactForm />)
})

it('should match the snapshot', () => {
  const { asFragment } = render(<ContactForm />)
  const html = asFragment()
  expect(html).toMatchSnapshot()
})

