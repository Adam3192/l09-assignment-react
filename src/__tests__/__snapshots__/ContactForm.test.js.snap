// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="formBasicName"
      >
        Name
      </label>
      <input
        class="form-control"
        id="formBasicName"
        name="name"
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="formBasicEmail"
      >
        Email
      </label>
      <input
        class="form-control"
        id="formBasicEmail"
        name="email"
        type="email"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="formBasicPhone"
      >
        Phone
      </label>
      <input
        class="form-control"
        id="formBasicPhone"
        name="phone"
        type="text"
        value=""
      />
    </div>
    <button
      class="btn btn-primary"
      type="submit"
    >
      Add Contact
    </button>
  </form>
</DocumentFragment>
`;
